### PR TITLE
Added Buratino clock

### PR DESCRIPTION
**Pull request by olegdvd**
Injected Clock to GatheredData
Removed unused
Added JPA test for GatheredSensorDataRepository

### CODE REVIEW

File: src/main/java/com/olegdvd/temperature/domain/GatheredSensorData.java [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.persistence.*;
+import java.time.Clock;
 import java.time.LocalDateTime;
-import java.util.UUID;
 
 @Entity
 public class GatheredSensorData {
```
Comment (kaidzen) [inline src/main/java/com/olegdvd/temperature/domain/GatheredSensorData.java:12]: As it entity, and also will be stored in collections for processing, should be added hashcode-equals contract

File: src/test/java/com/olegdvd/temperature/service/SensorRequestServiceTest.java [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
-import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.junit4.SpringRunner;
 
-import java.util.Optional;
-
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
```
Comment (kaidzen) [inline src/test/java/com/olegdvd/temperature/service/SensorRequestServiceTest.java:46]: For unit test any external dependencies should be mocked. In such condition it is raser end-to-end test, and should be @Ignore-d in unit tests

File: src/main/java/com/olegdvd/temperature/service/SensorRequestService.java [hunk 3]
```diff
@@ -24,12 +24,14 @@
     private final RestTemplate restTemplate;
     private final InfoRepository<SensorInfo> repository;
     private final GatheredSensorDateRepository gatheredSensorDateRepo;
+    private final Clock clock;
 
     @Autowired
-    public SensorRequestService(RestTemplate restTemplate, SensorInfoLocalRepository repository, GatheredSensorDateRepository gatheredSensorDateRepo) {
+    public SensorRequestService(RestTemplate restTemplate, SensorInfoLocalRepository repository, GatheredSensorDateRepository gatheredSensorDateRepo, Clock clock) {
```
Comment (kaidzen) [inline src/main/java/com/olegdvd/temperature/service/SensorRequestService.java:30]: Unit test should be added.

**Review by kaidzen — CHANGES_REQUESTED** [review]
Process it, please

File: src/main/java/com/olegdvd/temperature/domain/GatheredSensorData.java (hunk 1 above)
Comment (olegdvd) [reply src/main/java/com/olegdvd/temperature/domain/GatheredSensorData.java:12]: Implemented

File: src/main/java/com/olegdvd/temperature/service/SensorRequestService.java (hunk 3 above)
Comment (olegdvd) [reply src/main/java/com/olegdvd/temperature/service/SensorRequestService.java:30]: Done.

File: src/test/java/com/olegdvd/temperature/service/SensorRequestServiceTest.java (hunk 2 above)
Comment (olegdvd) [reply src/test/java/com/olegdvd/temperature/service/SensorRequestServiceTest.java:46]: Removed test that process with real dependancy.

**Review by kaidzen — APPROVED** [review]
Good!

**Pull request merged**